api/redhatopenshift/HcpCluster.Management/hcpCluster-models.tsp

```diff
@@ -456,7 +459,35 @@ model NodePoolAutoScaling {
   @minValue(0)
   max?: int32;
 }
+/** Disk encryption settings for the OS and data disks */
+model Encryption {
+  /** Infrastructure-level encryption for the disks. Defaults to "disabled". */
+  infrastructureEncryption?: "enabled" | "disabled" | string = "disabled";
+
+  /** Configuration for using a customer-managed key (CMK) for disk encryption. */
+  customerManagedKeyEncryption?: CustomerManagedKeyEncryption;
+}
 
+/** Configuration for customer-managed disk encryption using Azure Key Vault. */
+model CustomerManagedKeyEncryption {
+  /** Identity used to authenticate with the Azure Key Vault for CMK. */
+  keyEncryptionKeyIdentity: KeyEncryptionKeyIdentity;
```

Comment:
As we mentioned through other channels, this does not seem to be the interface that configuration of OS Disks SSE disk encryption with customer managed keys seems to expect.
Looking at https://learn.microsoft.com/en-us/azure/virtual-machines/disks-enable-customer-managed-keys-portal, this seems to be done through disk encryption sets.

You can also observe how hypershift seems to be exposing it: https://github.com/openshift/hypershift/blob/main/api/hypershift/v1beta1/azure.go#L315